python/src/aiconfig/editor/client/src/components/prompt/prompt_input/schema_renderer/PromptInputDataSchemaRenderer.tsx

```diff
@@ -14,12 +15,15 @@ export default memo(function PromptInputDataSchemaRenderer({
   data,
   onChangeData,
 }: Props) {
+  const {readOnly} = useContext(AIConfigContext);
+  
   switch (schema.type) {
     case "string":
       return (
         <Textarea
           value={data ? (data as string) : ""}
           onChange={(e) => onChangeData(e.target.value)}
+          disabled={readOnly}
           placeholder="Type a prompt"
```

Comment:
should this say something else in readonly mode?

Reply:
Placeholder is a value for when there is no data already and asking user to do an action. In our case let's set this to "" in readOnly mode (can you test example where the input string is empty) for both readOnly and normal? Thanks!

Reply:
placeHolder no longer needed since spoiler will render input which will be an undefined/empty string